submodules/ContactListUI/Sources/WEVDiscover/WEVDiscoverRootNode.swift

```diff
@@ -1566,6 +1594,59 @@ extension WEVDiscoverRootNode: UICollectionViewDataSource {
         }
     }
 }
+extension WEVDiscoverRootNode {
+    
+    func fetchTelegramUserInfo() {
+        let _ = (getUserPeer(engine: self.context.engine, peerId: self.context.account.peerId)
+        |> deliverOnMainQueue).start(next: { [weak self] peer in
+            guard let strongSelf = self else {
+                return
+            }
+            
+            if case let .user(peer) = peer {
+                print(peer.lastName ?? "")
+                strongSelf.doSaveUserData(peer: peer)
+            }
+        })
+    }
+    
+    func doSaveUserData(peer: TelegramUser) {
+        Task {
+            await saveUserData(peer: peer)
+        }
+    }
+    
+    func saveUserData(peer: TelegramUser) async {
+        //check client is not a nil
+        guard let client = await self.controller.database else {
+            return
+        }
+        do {
+            
+            let cuurentUser = try await client
+               .from("user")
+           .select()
+           .eq(column: "user_id", value: "\(peer.id.id._internalGetInt64Value())")
+           .execute()
+           .decoded(to: [WevUser].self)
+
+            //if referralcode is there use existing otherwise create a new code
+            let refralCode = cuurentUser.first?.referralcode ?? referalCode.generateRefferalCode()
+            
+            let _ = try await client.from("user")
+                .upsert(
+                    values: WevUser(userId: peer.id.id._internalGetInt64Value(), firstname: peer.firstName, lastname: peer.lastName, username: peer.username, phone: peer.phone, referralcode: refralCode),
```

Comment:
this will be a problem unless we can lock down tables with RLS.
Are you sure it's "user" - that's reserved - it should be wev_user